### PR TITLE
Preparation for OmemoKeys ViewController

### PR DESCRIPTION
**Pull request by matthewrfennell**
These are all purely refactors, there should be no change in behaviour.
Main purpose is to simplify `OmemoKeys` where there is logic that can be pushed out of the view, or removing state where we can use functions instead.

### CODE REVIEW

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
Please change `MLContact` so that it has an `account` property that does what `getAccountBelongingToContact` does. This is cleaner than an aux function in Swift. And it has the nice side effect that it can also be used from objc.

Please update `MLCall` to use this new property of `MLContact` in its `-(xmpp*) account` method.

BTW: `getAccountBelongingToContact` should have been in `SwiftHelpers.swift` instead of `SwiftuiHelpers.swift`.

File: Monal/Classes/MemberList.swift [hunk 1]
```diff
@@ -30,7 +30,7 @@ struct MemberList: View {
     @StateObject private var overlay = LoadingOverlayState()
 
     init(mucContact: ObservableKVOWrapper<MLContact>) {
-        account = MLXMPPManager.sharedInstance().getConnectedAccount(forID: mucContact.accountId)! as xmpp
+        account = mucContact.obj.account! as xmpp
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MemberList.swift:33]: You only have to use `.obj` if you are accessing a method of an `ObservableKVOWrapper` wrapped objc object, accessing a property of such an object (without n`.obj`) works just fine. And it has the additional benefit of emitting SwiftUI `ObservableObject` signals on KVO changes of that property in objc land.

**Review by tmolitor-stud-tu — APPROVED** [review]
I marked a minor nitpick. I saw you using `.obj` only in `.init` methods, so that's probably fine, but for clarity and to counter copy-paste bugs later on, this should be changed.

I'll merge it nonetheless, you can open a new PR for that minor thing :)

File: Monal/Classes/MemberList.swift (hunk 1 above)
Comment (matthewrfennell) [reply Monal/Classes/MemberList.swift:33]: Is there something special I have to do to allow it? When I try this I get "Generic parameter 'T' could not be inferred".

screenshot-1

**Comment by tmolitor-stud-tu** [comment]
You have to cast it for Swift's type system to be able to infer the type of the generic parameter...
An 'as! xmpp' should suffice :)

Am 6. August 2024 09:56:39 MESZ schrieb Matthew Fennell ***@***.***>:

**Pull request merged**